gch4i/proxy_processing/task_ng_trans_comp_station_proxy.py

```diff
@@ -0,0 +1,385 @@
+                                GHGRP_temp_data.loc[iplant, 'Env_name'] = enverus_notmatched.loc[list_envmat_filter.index[idxlat], 'facility_name']
+                                GHGRP_temp_data.loc[iplant, 'Env_county'] = enverus_notmatched.loc[list_envmat_filter.index[idxlat], 'county']
+                                GHGRP_temp_data.loc[iplant, 'Env_state'] = enverus_notmatched.loc[list_envmat_filter.index[idxlat], 'state_code']
+                                GHGRP_temp_data.loc[iplant, 'Env_fuel'] = enverus_notmatched.loc[list_envmat_filter.index[idxlat], 'fuel']
+                                rows_to_delete = np.append(rows_to_delete, list_envmat_filter.index[idxlat])
+                            # Assign to the plant with the closest lon
+                            elif vallat > vallon:
+                                GHGRP_temp_data.loc[iplant, 'match_flag'] = 1
+                                GHGRP_temp_data.loc[iplant, 'Env_name'] = enverus_notmatched.loc[list_envmat_filter.index[idxlon], 'facility_name']
+                                GHGRP_temp_data.loc[iplant, 'Env_county'] = enverus_notmatched.loc[list_envmat_filter.index[idxlon], 'county']
+                                GHGRP_temp_data.loc[iplant, 'Env_state'] = enverus_notmatched.loc[list_envmat_filter.index[idxlon], 'state_code']
+                                GHGRP_temp_data.loc[iplant, 'Env_fuel'] = enverus_notmatched.loc[list_envmat_filter.index[idxlon], 'fuel']
+                                rows_to_delete = np.append(rows_to_delete, list_envmat_filter.index[idxlon])
+
+    # Ensure rows to delete are integers
+    rows_to_delete = rows_to_delete.astype(int)
+    # Copy temp data to GHGRP facility emissions
+    ghgrp_facility_emissions = GHGRP_temp_data.copy()
+
+    # Matched and not matched GHGRP and Enverus data sets
+    # Unmatched Enverus plants
+    enverus_notmatched = enverus_notmatched.drop(rows_to_delete)
+    # Unique GHGRP plants
+    unique_GHGRP_plants = ghgrp_facility_emissions['facility_id'].unique()
+    # Unmatched GHGRP plants
+    unmatched_GHGRP_plants = ghgrp_facility_emissions[ghgrp_facility_emissions['match_flag'] == 0]
+    # Unique unmatched GHGRP plants
+    unique_unmatched_GHGRP_plants = unmatched_GHGRP_plants['facility_id'].unique()
+    # Matched GHGRP plants
+    matched_GHGRP_plants = GHGRP_temp_data[GHGRP_temp_data['match_flag'] == 1].reset_index(drop=False)
+
+    # QA/QC number of plants matched
+    print('QA/QC: Number of GHGRP plants not in Enverus data set')
+    print(len(unique_unmatched_GHGRP_plants), ' of ', len(unique_GHGRP_plants))
+    print('QA/QC: Number of Enverus plants not in GHGRP data set')
+    print(len(enverus_notmatched), ' of ', len(enverus_stations))
+
+    # Calculate the average emissions/fuel ratio for matched plants
+    matched_GHGRP_plants['emis_fuel_ratio'] = 0.0
+    for iplant in np.arange(0, len(matched_GHGRP_plants)):
```

Comment:
This looks like it could just be a vectorized operation since it iterates through rows? Or an apply since it has to be a function with if/else?
```suggestion
    matched_GHGRP_plants['emis_fuel_ratio'] = (matched_GHGRP_plants['ch4_emi'] / matched_GHGRP_plants['Env_fuel'] if matched_GHGRP_plants['Env_fuel'] > 0 else 0)
```